google-cloud-firestore/src/test/java/com/google/cloud/firestore/TransactionTest.java

```diff
@@ -132,6 +134,35 @@ public String updateCallback(Transaction transaction) {
     assertEquals(commit(TRANSACTION_ID), requests.get(1));
   }
 
+  @Test
+  public void returnsValueAsync() throws Exception {
+    doReturn(beginResponse())
+        .doReturn(commitResponse(0, 0))
+        .when(firestoreMock)
+        .sendRequest(requestCapture.capture(), Matchers.<UnaryCallable<Message, Message>>any());
+
+    ApiFuture<String> transaction =
+        firestoreMock.runAsyncTransaction(
+            new Transaction.AsyncFunction<String>() {
+              @Override
+              public ApiFuture<String> updateCallback(Transaction transaction) {
+                Assert.assertEquals("user_provided", Thread.currentThread().getName());
+                SettableApiFuture<String> apiFuture = SettableApiFuture.create();
+                apiFuture.set("foo");
+                return apiFuture;
```

Comment:
Nit: Replace with https://github.com/googleapis/api-common-java/blob/master/src/main/java/com/google/api/core/ApiFutures.java#L107 (here and below)

Reply:
replaced both places with ApiFutures#immediateFuture()